react/features/connection-indicator/components/ConnectionIndicator.js

```diff
@@ -0,0 +1,282 @@
+        this.popover = new JitsiPopover($(this._rootElement), {
+            content: this._renderStatisticsTable(),
+            skin: 'black',
+            position: interfaceConfig.VERTICAL_FILMSTRIP ? 'left' : 'top'
+        });
+
+        this.popover.addOnHoverPopover(this.props.onHover);
+    }
+
+    /**
+     * Updates the contents of the popover. This is done manually because the
+     * popover is not a React Component yet and so is not automatiucally aware
+     * of changed data.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentDidUpdate() {
+        this.popover.updateContent(this._renderStatisticsTable());
+    }
+
+    /**
+     * Cleans up any popover instance that is linked to the component.
+     *
+     * @inheritdoc
+     * returns {void}
+     */
+    componentWillUnmount() {
+        this.popover.forceHide();
+        this.popover.remove();
+    }
+
+    /**
+     * Implements React's {@link Component#render()}.
+     *
+     * @inheritdoc
+     * @returns {ReactElement}
+     */
+    render() {
+        return (
+            <div
+                className = 'connection-indicator indicator'
+                ref = { this._setRootElement }>
+                <div className = 'connection indicatoricon'>
+                    { this._renderIcon() }
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Callback to invoke when the show more link in the popover content is
+     * clicked. Sets the state which will determine if the popover should show
+     * additional statistics about the connection.
+     *
+     * @returns {void}
+     */
+    _onToggleShowMore() {
+        this.setState({ showMoreStats: !this.state.showMoreStats });
+    }
+
+    /**
+     * Creates a ReactElement for displaying an icon that represents the current
+     * connection quality.
+     *
+     * @returns {ReactElement}
+     */
+    _renderIcon() {
+        switch (this.props.stats.connectionStatus) {
+        case JitsiParticipantConnectionStatus.INTERRUPTED:
+            return (
+                <span className = 'connection_lost'>
+                    <i className = 'icon-connection-lost' />
+                </span>
+            );
+        case JitsiParticipantConnectionStatus.INACTIVE:
+            return (
+                <span className = 'connection_ninja'>
+                    <i className = 'icon-ninja' />
+                </span>
+            );
+        default: {
+            const { percent } = this.props.stats;
+            const width = QUALITY_TO_WIDTH.find(x => percent >= x.percent);
+
+            return [
+                <span
+                    className = 'connection_empty'
+                    key = 'icon-empty'>
+                    <i className = 'icon-connection' />
+                </span>,
+                <span
+                    className = 'connection_full'
+                    key = 'icon-full'
+                    style = {{ width: width && width.width }}>
```

Comment:
It's handled by react, plus browsers would know not to honor the style. Here's the line used by react that would cause null/undefined to no-op: https://github.com/facebook/react/blob/29eb21dd0496cc7312b9c383d414da37750a8dfa/src/renderers/dom/shared/dangerousStyleValue.js#L39.

However, I've made the change to be more defensive of this within jitsi-meet, just in case there's a change someday in the third-party libs.